review/comment/comment.go

```diff
@@ -126,9 +126,16 @@ type Comment struct {
 // New returns a new comment with the given description message.
 //
 // The Timestamp and Author fields are automatically filled in with the current time and user.
-func New(author string, description string) Comment {
+func New(author string, description string, date *time.Time) Comment {
```

Comment:
With this change (and the corresponding one in the `request` package), any clients of the API outside of this repository might get broken (we haven't migrated to using go modules yet).

I'm also wondering if the `New` methods actually provide any value in the case of the timestamp being explicitly provided (as opposed to clients manually creating the `Comment` struct).

The only benefit it seems to have is normalizing how a *time.Time object is converted to a timestamp.

Maybe a better place for that formatting helper is inside of the new `commands/helper` package that you've created?

I.E. the caller could invoke something like:

```go
	date, err := GetDate(*commentDate)
	if err != nil {
		return err
	}
	if date == nil {
		date = time.Now().Unix()
	}
	ts := FormatDate(date)
	c := comment.Comment{
		Timestamp: ts,
		Author: userEmail,
		Description: *commentMessage
	}
```

Alternatively, we could keep the `New` call as-is and just conditionally populate the timestamp after (similarly to how we populate location, parent, etc...).

```go
	c := comment.New(userEmail, *commentMessage)
	date, err := GetDate(*commentDate)
	if err != nil {
		return err
	}
	if date != nil {
		c.Timestamp = FormatDate(date)
	}
```

I don't have a strong preference of one approach over the other, but the latter does look like it would be a (slightly) smaller change.

Reply:
ok